plugins/dialog/plugin.js

```diff
@@ -1488,6 +1500,65 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 			}
 
 			this.fire( 'state', state );
+		},
+
+		/**
+		 * @inheritdoc CKEDITOR.dialog.modeledDialog#getModel
+		 */
+		getModel: function( editor ) {
+			var ret = this._.model;
+
+			// Always return cached model instance.
+			if ( this._.model ) {
+				return this._.model;
+			}
+
+			if ( this.definition.getModel ) {
+				ret = this.definition.getModel( editor ) || null;
+			}
+
+			if ( !ret ) {
+				ret = this.fire( 'getModel', {
+					model: ret
+				}, editor ).model;
+			}
+
+			return ret || null;
+		},
+
+		/**
+		 *
+		 * @since 4.11.0
+		 * @param {CKEDITOR.editor} editor
+		 * @param {CKEDITOR.dom.element/}
+		 */
+		setModel: function( editor, newModel ) {
+			this._.model = newModel;
+		},
+
+		/**
+		 * Method tells whether the dialog is editing an existing element or adding a new one.
+		 *
+		 * In case if dialog does not
+		 *
+		 * @since 4.11.0
+		 * @param {CKEDITOR.editor} editor
+		 * @returns {Boolean} Returns `true` if dialog is editing content that already exists in the editor.
+		 */
+		isEditing: function( editor ) {
+			var model = this.getModel( editor ),
+				ret = null;
+
+			if ( model && model instanceof CKEDITOR.dom.node ) {
+				ret = !!model.getParent();
+			}
+
+			ret = this.fire( 'isEditing', {
+				returnValue: ret,
+				model: model
+			} ).returnValue;
```

Comment:
Maybe just `editing` instead of `returnValue`?

Reply:
See https://github.com/ckeditor/ckeditor-dev/pull/2488#discussion_r225732434